core/src/components/datetime/datetime.tsx

```diff
@@ -1262,9 +1262,63 @@ export class Datetime implements ComponentInterface {
        */
       this.activeParts = [];
     }
+
+    /**
+     * Only animate if:
+     * 1. We're using grid style (wheel style pickers should just jump to new value)
+     * 2. The month and/or year actually changed (otherwise there's nothing to animate to)
+     * 3. The datetime is visible (prevents animation when in collapsed datetime-button, for example)
+     * 4. The month/year picker is not open (since you wouldn't see the animation anyway)
+     */
+    const didChangeMonth = month !== workingParts.month || year !== workingParts.year;
+    const elIsVisible = el.offsetParent !== null;
+    const { isGridStyle, showMonthAndYear } = this;
+    if (animate && isGridStyle && didChangeMonth && elIsVisible && !showMonthAndYear) {
+      /**
+       * Tell other render functions that we need to force the
+       * target month to appear in place of the actual next/prev month.
+       * Because this is a State variable, a rerender will be triggered
+       * automatically, updating the rendered months.
+       */
+      this.forceRenderDate = targetValue;
+
+      /**
+       * Flag that we've started scrolling to the forced date.
+       * The resolve function will be called by the datetime's
+       * scroll listener when it's done updating everything.
+       * This is a replacement for making prev/nextMonth async,
+       * since the logic we're waiting on is in a listener.
+       */
+      const forceDateScrollingPromise = new Promise<void>((resolve) => {
+        this.resolveForceDateScrolling = resolve;
+      });
+
+      /**
+       * Animate smoothly to the forced month. This will also update
+       * workingParts and correct the surrounding months for us.
+       */
+      const targetMonthIsBefore = isBefore(targetValue, workingParts);
+      targetMonthIsBefore ? this.prevMonth() : this.nextMonth();
+      await forceDateScrollingPromise;
```

Comment:
So I'm wondering if this should be pulled out into its own function. I'm not sure it makes sense to have `processValue` return a promise now. The purpose of that function is to grab `this.value` and process it into `DatetimeParts` (or warn if it's an invalid date). I don't think it should necessarily wait for any scrolling operations to finish since that's not required to process an ISO-8601 date.

The benefit of moving it to its own function is `processValue` doesn't need to be async anymore, but the new function cal still wait for scrolling to finish.

Reply:
Good call, I'll get this rearranged 👍